### PR TITLE
[expo] suppress `Overwriting fontFamily style attribute preprocessor`

### PR DESCRIPTION
**Pull request by Kudo**
# Why

when app startup, there's a yellow box warning about `Overwriting fontFamily style attribute preprocessor`. this is coming from upstream change: https://github.com/facebook/react-native/commit/33b385825c72
close ENG-4704

# How

`StyleSheet.setStyleAttributePreprocessor` is an experimented feature. it makes sense for upstream to show the warning. to make an expo app work as it is, i'll have to monkey patch `console.warn` temporarily to disable the warning.

# Test Plan

iOS unversioned expo go + NCL

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by Kudo** [comment]
sdk test is failed from #17091. we could address it in separated pr.

**Review by tsapeta — APPROVED** [review]

File: packages/expo/src/Expo.fx.tsx [hunk 1]
```diff
@@ -22,7 +22,17 @@ const isManagedEnvironment =
...
 if (StyleSheet.setStyleAttributePreprocessor) {
-  StyleSheet.setStyleAttributePreprocessor('fontFamily', Font.processFontFamily);
+  if (__DEV__) {
+    // Temporarily disable console.warn() in dev mode,
+    // because the experimented `StyleSheet.setStyleAttributePreprocessor` will show a warning about
+    // `Overwriting fontFamily style attribute preprocessor`.
+    const originalConsoleWarn = global.console.warn;
+    global.console.warn = () => {};
+    StyleSheet.setStyleAttributePreprocessor('fontFamily', Font.processFontFamily);
+    global.console.warn = originalConsoleWarn;
```
Comment (EvanBacon) [inline packages/expo/src/Expo.fx.tsx:32]: Seems fine, let me know if this changes the stack traces though. If it does we'll need to add a file matcher to [this](https://github.com/expo/expo-cli/blob/ae23775587df6b4c69967b3e49e8f6d0d95d489b/packages/metro-config/src/ExpoMetroConfig.ts#L20).

**Review by EvanBacon — APPROVED** [review]

**Review by brentvatne — COMMENTED** [review]
this is an OK workaround for now but it will be a problem in apps that use expo-font outside of the context where Expo.fx is executed. i think a regression was actually introduced in https://github.com/facebook/react-native/commit/33b385825c72 - the original intent of the warning was to only display it when the style preprocessor for a given style was already set to a function because the new value would be overwriting the existing processor

**Comment by Kudo** [comment]
i've confirmed the change doesn't impact to stacktrace and added a [linear task](https://linear.app/expo/issue/ENG-4745/propose-a-pr-to-react-native-for-fixing) to propose a pr to upstream later. let's merge the workaround first for sdk 45 release.

**Pull request merged**